### ISSUE TITLE
https://pub.dev/packages/wakelock_plus

### ISSUE THREAD

**Issue by syleishere** [flet-dev/flet#2999]
People using audioplayers package cannot get a wakelock on android to continue playing music, soon as screen turns off or app is backgrounded.

The flutter package https://pub.dev/packages/wakelock_plus seems to solve this issue, the video player would need it as well. Modifications would be needed to AndroidManifest.xml to add the permission.

I believe it would need a partial wakelock for audio, and full wakelock for video.

**Comment by ndonkoHenri** [comment]
I think `Video` already has wakelock: https://flet.dev/docs/controls/video#wakelock

**Comment by syleishere** [comment]
I believe the reason it may not be working in video is because AndroidManifest.xml does not include permission for it when I tried before. I haven't been able to test video on android lately because of a recent gradle upgrade on android studio that is breaking the  --include-packages flet_video:

```
e: Incompatible classes were found in dependencies. Remove them from the classpath or use '-Xskip-metadata-version-check' to suppress errors
e: C:/Users/dan/.gradle/caches/transforms-3/6498ee1438a7a83cb4725beb055db98c/transformed/jetified-kotlin-stdlib-1.9.23.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with
an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.
e: C:/Users/dan/.gradle/caches/transforms-3/6498ee1438a7a83cb4725beb055db98c/transformed/jetified-kotlin-stdlib-1.9.23.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with
an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.
e: C:/Users/dan/.gradle/caches/transforms-3/6498ee1438a7a83cb4725beb055db98c/transformed/jetified-kotlin-stdlib-1.9.23.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an
incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.
e: C:\Users\dan\AppData\Local\Pub\Cache\hosted\pub.dev\wakelock_plus-1.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\wakelock\Messages.g.kt: (14, 10): Unresolved reference: listOf
e: C:\Users\dan\AppData\Local\Pub\Cache\hosted\pub.dev\wakelock_plus-1.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\wakelock\Messages.g.kt: (19, 12): Unresolved reference: listOf
e: C:\Users\dan\AppData\Local\Pub\Cache\hosted\pub.dev\wakelock_plus-1.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\wakelock\Messages.g.kt: (25, 12): Unresolved reference: listOf
e: C:\Users\dan\AppData\Local\Pub\Cache\hosted\pub.dev\wakelock_plus-1.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\wakelock\Messages.g.kt: (26, 17): Unresolved reference: javaClass
e: C:\Users\dan\AppData\Local\Pub\Cache\hosted\pub.dev\wakelock_plus-1.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\wakelock\Messages.g.kt: (55, 6): Class 'kotlin.Suppress' was
compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.
The class is loaded from C:/Users/dan/.gradle/caches/transforms-3/6498ee1438a7a83cb4725beb055db98c/transformed/jetified-kotlin-stdlib-1.9.23.jar!/kotlin/Suppress.class
e: C:\Users\dan\AppData\Local\Pub\Cache\hosted\pub.dev\wakelock_plus-1.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\wakelock\Messages.g.kt: (62, 12): Unresolved reference: listOf
e: C:\Users\dan\AppData\Local\Pub\Cache\hosted\pub.dev\wakelock_plus-1.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\wakelock\Messages.g.kt: (78, 6): Class 'kotlin.Suppress' was
compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.
```
Type errors everywhere, I tried to look for a fix and think the answer is here:
https://stackoverflow.com/questions/67699823/module-was-compiled-with-an-incompatible-version-of-kotlin-the-binary-version-o

Basically to modify build.grade for the project, not sure if you have run into this upgrading kotlin recently. Yeah I have to agree at this point in time, the video plugin would be a much better idea to build a music player with as it supports playlists, without playlist support, normal audio plugin will not be able to play more than one song on web when someone switches tabs in the browser. Only issue I have found with video is if I page.overlay it, I can't be switching routes the normal way with ft.view, I have to delete controls manually, then add new ones on route changes then, skipping ft.view completely, which is fine, be more like using tabs then.

**Comment by syleishere** [comment]
I still stand by my original post though, even if we can get video player working correctly in android. This plugin should be a separate plugin. In the future of flet people may develop apps that need screen on, say for map navigation in their car, background audio, cooking recipes whatever. A generic plugin for android that can act as a foreground service for everyone where they could put custom icons/controls/images on users display as well will be what is needed and what I think will make flet really stand out for developers. 

I think we might need a customized plugin that makes it easy for people to run a foreground service on android, maybe a function where they could add custom controls/icons/background images to users screen. I think that would make flet really stand out as a developers paradise and gain it huge popularity with developers, as any serious developers are going to target android and iphone before anything else.

**Comment by syleishere** [comment]
hey guys, is there any way you can wrap mediakit with https://pub.dev/packages/audio_service
I would be most grateful, https://crystaltunes.sunsaturn.com:9000/music  
this is project currently working on, I really need to finish project before June 26th and I cannot without that plugin otherwise it will be useless on android/iphone etc. Would need AndroidManifest.xml edits and plist for Iphone as well according to that page.
